state-chain/pallets/cf-elections/src/electoral_system.rs

```diff
@@ -397,3 +397,278 @@ pub use access::{
 	ConsensusStatus, ElectionReadAccess, ElectionWriteAccess, ElectoralReadAccess,
 	ElectoralWriteAccess,
 };
+
+#[cfg(test)]
+pub mod mocks {
+	use super::*;
+
+	use core::marker::PhantomData;
+	use std::collections::BTreeMap;
+
+	use crate::{
+		electoral_system::{ConsensusStatus, ElectionIdentifierOf, ElectoralReadAccess},
+		UniqueMonotonicIdentifier,
+	};
+
+	pub struct MockElectionReadAccess<'es, ES: ElectoralSystem> {
+		electoral_system: &'es MockElectoralAccess<ES>,
+		election_identifier: ElectionIdentifierOf<ES>,
+		_phantom: PhantomData<ES>,
+	}
+
+	pub struct MockElectionWriteAccess<'es, ES: ElectoralSystem> {
+		electoral_system: &'es mut MockElectoralAccess<ES>,
+		election_identifier: ElectionIdentifierOf<ES>,
+		_phantom: PhantomData<ES>,
+	}
+
+	macro_rules! impl_election_read_access {
+		($struct_name:ident) => {
+			impl<ES: ElectoralSystem> ElectionReadAccess for $struct_name<'_, ES> {
+				type ElectoralSystem = ES;
+
+				fn settings(
+					&self,
+				) -> Result<
+					<Self::ElectoralSystem as ElectoralSystem>::ElectoralSettings,
+					CorruptStorageError,
+				> {
+					Ok(self
```

Comment:
I see you are panicking, I'd suggest returning the CorruptStorage error instead in all thsese mocks as that is the behaviour of the pallet.